### ISSUE TITLE
[Logistics] [0007] Setup ambushed mission and connect with logistics state machine

### ISSUE THREAD

**Issue by mwpowellhtx** [mwpowellhtx/KP-Liberation#51]
### Basic Information
**Mission version:** `v0.98.dev-s8`
**Map used:** _Altis_

### Individual Things
[Following on](https://github.com/mwpowellhtx/KP-Liberation/issues/29), _ambushed_ logistics lines may occur during the _en route_ state `onStateEntered`. This potentially occurs when `civilian reputation` is _sufficiently negative_, or _hostile_, towards friendly forces.

When condition is met, _ambush_ occurs, the transports are created at or near the nearest road marker or open space, and crates dropped onto the field as well. Then, of course, the hostile civilian resistance is setup to patrol the area. All contingent on the civilian reputation in terms of their strength, skill, etc.

The logistics line is frozen in _ambushed_ status until the mission is resolved one way or another. Crates will be periodically removed from the target area. Additionally, transports may be progressively damaged during the same period. Resolution happens when all crates are fully destroyed, or players intervene to neutralize the hostile resistance.

Transports may remain for a period of time in whatever form they are left in when the mission is resolved. Subject to the same transient vehicle life cycle heuristics as the rest of the mission; i.e. may be GC'ed after a period of time, etc, etc, etc.

Open to details being negotiated differently, but this is the basic vision for the mission, logistics status, etc.

**Comment by mwpowellhtx** [comment]
Added `logamb` secondary objective. Entry point is the `trigger` function, in which case _semaphore_ is evaluated whether additional _children_ may be supported.

On the front side, we evaluate whether `canBeAmbushed`, which takes into consideration things CBA such as:
1. _Ambush interval_ expiry
1. CBA settings such as _Routes can be blocked_, _blocked routes can be ambushed_, _friendly zone ambushes_
1. Whether the current `LINE` in question is considered 'high value' i.e. there are no other lines with a higher _storage value_
1. There are _roads_ in proximity which may support the ambush
1. Last but not least, there is _civilian hostility_ sufficient to justify whether ambush may occur

**Comment by mwpowellhtx** [comment]
Concluding the effort, or at least introducing a viable rough cut first draft:

Mission version: `v0.98.dev-s31`

**Comment by mwpowellhtx** [comment]
Couple of remaining items to follow up on, but will file issues on this separately.
1. [_Correct patrol deployment for transports_](106)

We also want to keep an eye on _objective completion conditions_. We think we have sorted them out in the current draft, but we can always learn new things under various play testing scenarios.

Closing for now, will follow up in separate issues as the need arises.
